src/functions/anti-scam/checkScam.ts

```diff
@@ -1,33 +1,49 @@
 import type { Redis } from 'ioredis';
 import { container } from 'tsyringe';
+import { URL } from 'node:url';
 
 import { logger } from '../../logger';
 import { kRedis } from '../../tokens';
 import { resolveRedirect } from '../../util/resolveRedirect';
 
+function urlOption(url: string): URL | null {
+	try {
+		return new URL(url);
+	} catch {
+		return null;
+	}
+}
+
 export async function checkScam(content: string): Promise<string[]> {
 	const redis = container.resolve<Redis>(kRedis);
 
-	const urlRegex =
-		/(http(s)?:\/\/.)?(www\.)?[-a-zA-Z0-9@:%._\+~#=]{2,256}\.[a-z]{2,6}\b([-a-zA-Z0-9@:%_\+.~#?&//=]*)/gi;
+	const linkRegex = /(?:https?:\/\/)(?:www\.)?[-a-zA-Z0-9@:%._\+~#=]{2,256}\.[a-z]{2,6}\b[-a-zA-Z0-9@:%_\+.~#?&//=]*/gi;
 	const scamDomains = await redis.smembers('scamdomains');
 	const trippedDomains = [];
 
 	let matches: any[] | null = [];
-	while ((matches = urlRegex.exec(content)) !== null) {
-		const url = matches[0];
-		const urlLower = url.toLowerCase();
-		const hit = scamDomains.find((d) => urlLower.includes(d.toLowerCase()));
+	while ((matches = linkRegex.exec(content)) !== null) {
```

Comment:
Given that the built-in URL class has a more robust URL parser than this regex, what do you think about splitting on whitespace and parsing each word? Do we think there might be any performance gains?

Reply:
while scammers don't currently do it, they could invalidate the URL scanner with just a prefix while keeping the link clickable in discord

Reply:
That's unfortunate to hear. Carry on!